openpype/hosts/max/plugins/publish/extract_tycache.py

```diff
@@ -0,0 +1,197 @@
+import os
+
+import pyblish.api
+from pymxs import runtime as rt
+
+from openpype.hosts.max.api import maintained_selection
+from openpype.pipeline import publish
+
+
+class ExtractTyCache(publish.Extractor):
+    """
+    Extract tycache format with tyFlow operators.
+    Notes:
+        - TyCache only works for TyFlow Pro Plugin.
+
+    Args:
+        self.export_particle(): sets up all job arguments for attributes
+            to be exported in MAXscript
+
+        self.get_operators(): get the export_particle operator
+
+        self.get_files(): get the files with tyFlow naming convention
+            before publishing
+    """
+
+    order = pyblish.api.ExtractorOrder - 0.2
+    label = "Extract TyCache"
+    hosts = ["max"]
+    families = ["tycache"]
+
+    def process(self, instance):
+        # TODO: let user decide the param
+        start = int(instance.context.data.get("frameStart"))
+        end = int(instance.context.data.get("frameEnd"))
+        self.log.info("Extracting Tycache...")
+
+        stagingdir = self.staging_dir(instance)
+        filename = "{name}.tyc".format(**instance.data)
+        path = os.path.join(stagingdir, filename)
+        filenames = self.get_files(instance, start, end)
+        additional_attributes = instance.data.get("tyc_attrs", {})
+
+        with maintained_selection():
+            job_args = None
+            has_tyc_spline = (
+                True
+                if instance.data["tycache_type"] == "tycachespline"
+                else False
+            )
```

Comment:
This is a little bit simpler:

```suggestion
            has_tyc_spline = instance.data["tycache_type"] == "tycachespline"
```